compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -4272,31 +4275,104 @@ declare_lint! {
 }
 
 declare_lint! {
-    /// The `unknown_or_malformed_diagnostic_attributes` lint detects unrecognized or otherwise malformed
-    /// diagnostic attributes.
+    /// The `malformed_diagnostic_attributes` lint detects malformed diagnostic attributes.
     ///
     /// ### Example
     ///
     /// ```rust
-    /// #![feature(diagnostic_namespace)]
-    /// #[diagnostic::does_not_exist]
-    /// struct Foo;
+    /// #[diagnostic::do_not_recommend(message = "message")]
+    /// trait Trait {}
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// It is usually a mistake to use options or syntax that is not supported. Check the spelling,
+    /// and check the diagnostic attribute listing for the correct name and syntax. Also
+    /// consider if you are using an old version of the compiler; perhaps the option or syntax
+    /// is only available in a newer version. See the [reference] for a list of diagnostic attributes
+    /// and their syntax.
```

Comment:
```suggestion
    /// It is usually a mistake to use options or syntax that is not
    /// supported. Check the spelling, and check the diagnostic attribute
    /// listing for the correct name and syntax. Also consider if you are
    /// using an old version of the compiler; perhaps the option or syntax is
    /// only available in a newer version. See the [reference] for a list of
    /// diagnostic attributes and the syntax of each.
```

Probably it'd be better to wrap this mechanically.  Also I'd suggest a slight wording change here (see the last few words).

Reply:
What is the mechanical wrapping tool of which you speak?

Reply:
I don't think we should be doing forced 80 col wrapping, it makes the text much harder to edit with nice diffs in practice. If someone wants to write it like this they're welcome to, but I really don't think we should enforce this as a style.

Reply:
I also don't like 80col that much. I've wrapped to 100col instead.

Reply:
> What is the mechanical wrapping tool of which you speak?

Whatever width we want is OK; my point here and elsewhere is that it was wrapped inconsistently, and in some cases had exceeded 100 columns.